ip_country/src/country_finder.rs

```diff
@@ -0,0 +1,301 @@
+use crate::country_block_serde::{
+    CountryBlockDeserializer, CountryBlockDeserializerIpv4, CountryBlockDeserializerIpv6,
+};
+use crate::country_block_stream::{Country, CountryBlock};
+use lazy_static::lazy_static;
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+
+lazy_static! {
+    pub static ref COUNTRY_CODE_FINDER: CountryCodeFinder = CountryCodeFinder::new(
+        crate::dbip_country::ipv4_country_data(),
+        crate::dbip_country::ipv6_country_data()
+    );
+}
+pub struct CountryCodeFinder {
+    ipv4: Vec<CountryBlock>,
+    ipv6: Vec<CountryBlock>,
+}
+
+impl CountryCodeFinder {
+    pub fn new(ipv4_data: (Vec<u64>, usize), ipv6_data: (Vec<u64>, usize)) -> Self {
+        Self {
+            ipv4: Self::initialize_country_finder_ipv4(ipv4_data),
+            ipv6: Self::initialize_country_finder_ipv6(ipv6_data),
+        }
+    }
+
+    fn initialize_country_finder_ipv4(data: (Vec<u64>, usize)) -> Vec<CountryBlock> {
+        //TODO write two tests - one for deserializer and one for loop to fill up the vec
```

Comment:
I made this test `deserialize_country_blocks_ipv4_and_ipv6_adn_fill_vecs`. So do there have to be two tests instead of one? I was designing this test to simulate the process of loading the `CountryBock` objects to memory on the Node startup. So we have a pretty close idea of how this deserialization will affect the Node startup.